### ISSUE TITLE
[deepseek_v3] test_mla prefill killed at seq_len >=32768

### ISSUE THREAD

**Issue by yieldthought** [tenstorrent/tt-metal#34307]
Parent: https://github.com/tenstorrent/tt-metal/issues/34305

Environment
- Set `DEEPSEEK_V3_HF_MODEL` and `DEEPSEEK_V3_CACHE` for your machine.

Gotcha (required for long seq lens)
- `models/demos/deepseek_v3/conftest.py` fixture `hf_config_short` clamps `max_seq_len = 3 * 1024`; to run prefill `seq_len >= 4096`, temporarily bump that (e.g. set it to `hf_config.max_position_embeddings` / `131072`).

Failing seq lens (from sweep table)
- Fails (prefill): 32768, 65536, 131072
- Note: 4096/8192/16384 are marked as passing only with a local patch (conftest.py + synchronize).

Error
```
Process Killed (patch to conftest.py + synchronize)
```

Reproducer
```bash
# Use PR that expands seq_len coverage
gh pr checkout 34177

export MESH_DEVICE=TG

pytest models/demos/deepseek_v3/tests/test_mla.py -k "prefill and 32768" -vv --maxfail=1
```

**Comment by yalrawwashTT** [comment]
Failing on sequence length 32k tokens, testing with 1 user per row on TG.

Killed on line where it's setting up reference IO:

```
state_dict, position_ids, torch_input, reference_output, input_cache, output_cache = generate_reference_io(
        model_path, module_path, hf_config_short, layer_idx, seq_len, batch_size, mode, state_dict
    ) 
```

for both run_test_forward_pass_mla1d and run_test_forward_pass_mla2d.

**Comment by yalrawwashTT** [comment]
It's an issue with the reference implementation where attention mechanism is done in a naive way and depends on sequence length in O(seq_len^2). Takes too much DRAM. Working on solution.

**Comment by yalrawwashTT** [comment]
Using chunking in `run_reference_with_attention` resolved this, but another hang encountered in `MLA1D.forward_prefill(tt_input, user_id, cur_row_idx, run_config, tt_rope_tensors, tt_page_table)` at `ttnn.synchronize_device(cfg["mesh_device"])`

**Comment by yalrawwashTT** [comment]
Now passing for 32k tokens. The following changes were necessary:

`mla1d.py`:
Added sequence chunking (8192 tokens) before all_gather_async in forward_prefill to avoid hangs on large sequences (32K+). Reshapes [1, seq_len, heads, dim] to [num_chunks, chunk_size, heads, dim] for gather, then processes chunks through the linear layer before de-chunking.

`test_utils.py`:
Added chunked processing (8192 tokens) in run_reference_with_attention for sequences >8192 to avoid OOM for reference model attention mechanism (naive O(n^2)). Processes the sequence in chunks with proper causal masking, accumulating KV cache across chunks, and concatenates outputs at the end.

64k tokens + 128k tokens still in debug.

**Comment by yalrawwashTT** [comment]
Up until 32k tokens resolved for mla1d, but this module will no longer be investigated because it is only used by pipeline parallel model, which is no longer a focus or product of development.

**Comment by yalrawwashTT** [comment]
PR merged, addressing MLA 2d prefill functionality up to 128k tokens.
